python/cuml/multiclass/multiclass.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import cuml.internals
+import sklearn.multiclass
+
+from cuml.common.array import CumlArray
+from cuml.common.base import Base, ClassifierMixin
+from cuml.common import input_to_host_array
+
+
+class MulticlassClassifier(Base, ClassifierMixin):
+    """ Wrapper around scikit-learn multiclass classifiers that allows to
+    choose different multiclass strategies.
+
+    See issue https://github.com/rapidsai/cuml/issues/2876 for more info about
+    using Sklearn meta estimators.
+    """
+    def __init__(self, estimator, handle=None, verbose=False,
+                 output_type=None, n_jobs=None, strategy='ovr'):
+        super().__init__(handle=handle, verbose=verbose,
+                         output_type=output_type)
+        self.strategy = strategy
+        self.estimator = estimator
+        self.n_jobs = n_jobs
+
+    @property
+    @cuml.internals.api_base_return_array_skipall
+    def classes_(self):
+        return self.multiclass_estimator.classes_
+
+    @property
+    @cuml.internals.api_base_return_any_skipall
+    def n_classes_(self):
+        return self.multiclass_estimator.n_classes_
+
+    def fit(self, X, y) -> 'MulticlassClassifier':
+        if self.strategy == 'ovr':
+            self.multiclass_estimator = sklearn.multiclass.\
+                OneVsRestClassifier(self.estimator, n_jobs=self.n_jobs)
+        elif self.strategy == 'ovo':
+            self.multiclass_estimator = \
+                sklearn.multiclass.OneVsOneClassifier(
+                    self.estimator, n_jobs=self.n_jobs)
+        else:
+            raise ValueError('Invalid multiclass strategy ' +
+                             str(self.strategy) + ', must be one of '
+                             '{"ovr", "ovo"}')
+        X, _, _, _, _ = input_to_host_array(X)
+        y, _, _, _, _ = input_to_host_array(y)
+        with cuml.internals.exit_internal_api():
+            return self.multiclass_estimator.fit(X, y)
```

Comment:
Won't this cause a device to host (and then back to device) copy when using any gpu object for calling `fit`, since the `fit` then does `input_to_cuml_array` https://github.com/rapidsai/cuml/blob/10e3bdb66e771fccf4d4c8ef55b70c3b6934ba4b/python/cuml/svm/svc.pyx#L410? So, shouldn't these calls be `input_to_cuml_array`?

Reply:
Hi Dante, your observation is correct, it will cause D2H and H2D copies. We have discussed this in issue #2876. For SVM, the cost of the copies are small compared to the time to fit.

While the issue is linked here, probably the docstring of multiclass should highlight more prominantly that the data is copied back and fort.

Reply:
makes sense, so yeah just documenting it more prominently indeed for now would be good

Reply:
I have improved the doc.